compiler/daml-extension/src/extension.ts

```diff
@@ -147,6 +151,37 @@ async function showReleaseNotes(version: string) {
     });
 }
 
+// Check if there is a new blog post which the user has not yet seen.
+// If so, display a notification with the link to the new post.
+// Update the user state so we don't notify about the same blog post again.
+// The user can opt out of these notifications entirely by changing the
+// 'daml.showNewBlogPosts' setting.
+async function showBlogIfNotSeen(config: WorkspaceConfiguration, context: ExtensionContext) {
+    if (!config.get('showNewBlogPosts')) { return; }
+    const feedUrl = 'https://blog.daml.com/daml-driven/rss.xml';
+    fetch(feedUrl).then(async (res: Response) => {
```

Comment:
Shouldn’t we be using an `await` on the `fetch` here? Our typescript expert @hurryabit can probably tell you more :trollface:

Reply:
Hmm reading/thinking about this now

Reply:
So, `await` means that "the evaluation of the async function halts until the promise is resolved". Don't know why we would need to do this here - it seems like the notification might be slightly delayed this way but don't think that's an issue.

Reply:
Yes, let's please use await. That makes it easier to read.

Reply:
`.then` only gets called once the promise has been resolved too. `const x = await P; E` and `P.then(x => E)` are pretty much the same. IMO, the former is easier to read. It's a bit like using `do` syntax rather than `>>= \_ ->` in Haskell. (In fact, promise is a monad, probably modulo some quirks.)